### PR TITLE
Fix inventory sync when canceling InventoryClickEvent, closes #181

### LINKED ISSUE #181: Inventory sync issue
Because images say more then words. I dobble click on in the inventory,
gif http://i.imgur.com/EkdKMbW.gif

Console claims that it was rejected:

```
2016/01/25 22:31:14 [INFORMATION] Paulomart: [rejected] WindowClickMessage(id=7, slot=-999, button=0, transaction=4, mode=4, item=null)
2016/01/25 22:31:51 [INFORMATION] Paulomart: [rejected] WindowClickMessage(id=8, slot=3, button=0, transaction=1, mode=0, item=ItemStack{STAINED_GLASS_PANE x 1, UNSPECIFIC_META:{meta-type=UNSPECIFIC, display-name=§f}})
2016/01/25 22:31:51 [INFORMATION] Paulomart: [rejected] WindowClickMessage(id=8, slot=3, button=0, transaction=2, mode=6, item=null)
2016/01/25 22:31:53 [INFORMATION] Paulomart: [rejected] WindowClickMessage(id=8, slot=-999, button=2, transaction=5, mode=5, item=null)
2016/01/25 22:31:54 [INFORMATION] Paulomart: [rejected] WindowClickMessage(id=8, slot=66, button=0, transaction=6, mode=0, item=ItemStack{STAINED_GLASS_PANE x 10, UNSPECIFIC_META:{meta-type=UNSPECIFIC, display-name=§f}})
2016/01/25 22:31:54 [INFORMATION] Paulomart: [rejected] WindowClickMessage(id=8, slot=-999, button=2, transaction=17, mode=5, item=null)
2016/01/25 22:31:55 [INFORMATION] Paulomart: [rejected] WindowClickMessage(id=8, slot=85, button=0, transaction=18, mode=0, item=null)
2016/01/25 22:31:56 [INFORMATION] Paulomart: [rejected] WindowClickMessage(id=8, slot=49, button=0, transaction=19, mode=0, item=ItemStack{STAINED_GLASS_PANE x 1, UNSPECIFIC_META:{meta-type=UNSPECIFIC, display-name=§f}})
2016/01/25 22:31:56 [INFORMATION] Paulomart: [rejected] WindowClickMessage(id=8, slot=49, button=0, transaction=20, mode=6, item=null)
2016/01/25 22:31:57 [INFORMATION] Paulomart: [rejected] WindowClickMessage(id=8, slot=-999, button=2, transaction=25, mode=5, item=null)
```
## <bountysource-plugin>

Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/30200468-inventory-sync-issue?utm_campaign=plugin&utm_content=tracker%2F14691067&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F14691067&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by Paulomart**

### CODE REVIEW

File: src/main/java/net/glowstone/net/handler/play/inv/WindowClickHandler.java [hunk 1]
```diff
@@ -31,10 +32,11 @@ public void handle(GlowSession session, WindowClickMessage message) {
         } catch (IllegalArgumentException ex) {
             GlowServer.logger.warning(session.getPlayer().getName() + ": illegal argument while handling click: " + ex);
         }
+        session.send(new TransactionMessage(message.getId(), message.getTransaction(), result));
         if (!result) {
-            GlowServer.logger.info(session.getPlayer().getName() + ": [rejected] " + message);
+            session.send(new SetWindowSlotMessage(-1, -1, null));
```
Comment (Paulomart) [inline src/main/java/net/glowstone/net/handler/play/inv/WindowClickHandler.java]: Maybe use player::getItemOnCourser here?

**Comment by Paulomart** [comment]
So okey, this throws a Exception if you click in the botten inventory, but as [InventoryView](https://github.com/GlowstonePlusPlus/Glowkit/blob/bukkit%2Bglowkit/src/main/java/org/bukkit/inventory/InventoryView.java#L164) states, it gets the Item by its `raw` id. But glowstone++ Overrides this and calls `checkSlot` which throws the error.

**Comment by gdude2002** [comment]
This looks a little hacked together - Maybe @deathcap can take a look at this at some point

**Comment by Paulomart** [comment]
@gdude2002 this is very hacked, but CraftBukkit was doing it to, it's a problem on the client, not accepting the transaction doesn't reset the slot and the courser of the client. Likely because this can never happen on a vanilla server. Still there is the issue with the inventory view.

**Comment by mastercoms** [comment]
Shouldn't this be on [line 231](https://github.com/GlowstonePlusPlus/GlowstonePlusPlus/blob/master/src/main/java/net/glowstone/net/handler/play/inv/WindowClickHandler.java#L231)?

**Comment by Paulomart** [comment]
I figured that the other message was sent out there too, so I did it there.

**Comment by mastercoms** [comment]
But a lot of other conditions make result false other than cancelling the event. In the process method, you can also use the inv variable, which will possibly fix the exception shown above.

**Comment by Paulomart** [comment]
But then the reject message for the action would have to be sent before the setslot message

**Comment by mastercoms** [comment]
https://github.com/GlowstonePlusPlus/GlowstonePlusPlus/commit/1b0b37d37b58ec003e40b5e0bfc3fb40690d4d1f I've tested it and it works.

**Comment by Paulomart** [comment]
Okey then, but what is with the bottom inventory? Did you test that?

**Comment by mastercoms** [comment]
It doesn't return the exception, but now I have made it return false for the bottom inventory, in this [commit](https://github.com/GlowstonePlusPlus/GlowstonePlusPlus/commit/13df8e5553b00b71e8bd24c6b09a335433c6006b). Thank you for that catch.

**Comment by mastercoms** [comment]
Bottom inventory now handled in https://github.com/GlowstonePlusPlus/GlowstonePlusPlus/commit/710d891b8f2a487c752f43d4616ca2057b00d30c

**Pull request closed without merging**